### PR TITLE
Test hostname verification

### PR DESCRIPTION
**Pull request by raboof**
To show this use currently fails, it will succeed after
applying https://github.com/akka/akka/pull/26906

Refs  #27086

### CODE REVIEW

**Comment by raboof** [comment]
PLS BUILD

File: akka-remote/src/test/scala/akka/remote/artery/tcp/TlsTcpSpec.scala [hunk 1]
```diff
@@ -183,22 +185,65 @@ class TlsTcpWithHostnameVerificationSpec
...
+      val rootB = RootActorPath(addressB)
+
+      systemB.actorOf(TestActors.echoActorProps, "echo")
+      system.actorSelection(rootB / "user" / "echo") ! Identify("echo")
+      val id = expectMsgType[ActorIdentity]
+
+      id.ref.get ! "42"
+      expectMsg("42")
+
+      systemB.terminate()
```
Comment (johanandren) [inline akka-remote/src/test/scala/akka/remote/artery/tcp/TlsTcpSpec.scala]: Is there also a case where the client cert does not matter that should be covered given that `Hostname verification should be done by the client only`?

Comment (raboof) [reply akka-remote/src/test/scala/akka/remote/artery/tcp/TlsTcpSpec.scala]: This test fails when ran without #26906 (which has since been merged to master). It's a bit roundabout but shows the problem reported there... not sure if there's a more 'direct' way to demonstrate it.

Comment (johanandren) [reply akka-remote/src/test/scala/akka/remote/artery/tcp/TlsTcpSpec.scala]: So this actually does test that the server doesn't verify the client hostname? A bit hard to follow what is the client given p2p :)

File: akka-remote/src/test/scala/akka/remote/artery/tcp/TlsTcpSpec.scala [hunk 2]
```diff
@@ -183,22 +185,65 @@ class TlsTcpWithHostnameVerificationSpec
...
+          occurrences = 3)
+        .intercept {
+          system.actorSelection(rootB / "user" / "echo") ! Identify("echo")
+        }
       expectNoMessage(2.seconds)
+      systemB.terminate()
+    }
+    "succeed when the name in the server certificate matches" in {
+      if (!arteryTcpTlsEnabled())
+        pending
```
Comment (johanandren) [inline akka-remote/src/test/scala/akka/remote/artery/tcp/TlsTcpSpec.scala]: It sholdn't be run at all though, if tls not enabled, rather than pending, I think you could just wrap the whole test case in a `if (arteryTcpTlsEnabled)`.

**Review by johanandren — APPROVED** [review]

File: akka-remote/src/test/resources/Makefile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+all: truststore keystore
```
Comment (chbatey) [inline akka-remote/src/test/resources/Makefile]: Good to have this checked in

**Review by chbatey — APPROVED** [review]
LGTM
